packages/app/src/model/databaseManager.ts

```diff
@@ -180,7 +181,10 @@ export default class DatabaseManager {
       }
     }
 
-    await this.actionLogStore.saveActionLogs(entries);
+    // NOTE: As of Typescript 4.2.4 PromptActionLog is not equivalent ActionLog. Looks to be compiler limitation
```

Comment:
If possible, I'd love to confine this narrowing issue to our bottleneck function (`getActionLogFromPromptActionLog`), so that if the conversion does indeed involve some actual value re-mapping, the call-sites will all get the new types as appropriate. So could we make this something like?:

```
    await this.actionLogStore.saveActionLogs(
      entries.map(({log, id}) => ({ log: getActionLogFromPromptActionLog(log); id })),
    );
```

Reply:
I will make the change! 

I originally omitted using the bottle neck function since it seemed a bit wasteful from a performance standpoint to map through an array just to cast it for typescript. However, I guess the performance impact is likely negligible.

Reply:
Makes sense. If the JIT's doing its job, it may be inlined out! :)